core/src/main/java/org/apache/accumulo/core/metadata/schema/RootGcCandidatesJson.java

```diff
@@ -54,14 +50,16 @@ private static class GSonData {
    *
    * SortedMap<dir path, SortedSet<file name>>
    */
-  private SortedMap<String,SortedSet<String>> candidates;
+  private final SortedMap<String,SortedSet<String>> candidates;
 
-  public RootGcCandidates() {
+  public RootGcCandidatesJson() {
     this.candidates = new TreeMap<>();
   }
 
-  private RootGcCandidates(SortedMap<String,SortedSet<String>> candidates) {
-    this.candidates = candidates;
+  public RootGcCandidatesJson(String jsonString) {
+    var rootGcCandidatesJson = GSON.fromJson(jsonString, RootGcCandidatesJson.class);
+    Preconditions.checkArgument(rootGcCandidatesJson.getVersion() == 1);
```

Comment:
I think you want to compare it to the version above, as in:

```suggestion
    Preconditions.checkArgument(rootGcCandidatesJson.getVersion() == version);
```